src/sql-parser/src/parser.rs

```diff
@@ -1305,6 +1300,31 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_at_most_one_keyword(
+        &mut self,
+        keywords: &[Keyword],
+        location: &str,
+    ) -> Result<Option<Keyword>, ParserError> {
+        match self.parse_one_of_keywords(keywords) {
+            Some(first) => {
+                if let Some(second) = self.parse_one_of_keywords(keywords) {
```

Comment:
Realized this needs to account for two instances of the same keyword following each other, e.g.

```
let keywords = keywords.retain(|kw| kw != first);
```

The parser is largely performance insensitive so this isn't a big deal.

Also think we should add a test for this case, e.g. `DROP DATABASE foo CASCADE CASCADE`

Reply:
good catch